Core.UnitTests/Serialization/SerializationEventRaiserTest.cs

```diff
@@ -78,20 +78,21 @@ public void InvokeAttributedMethod_UsesCache ()
       ClassWithDeserializationEvents instance = new ClassWithDeserializationEvents ();
       StreamingContext context = new StreamingContext();
 
-      MockRepository repository = new MockRepository();
-      SerializationEventRaiser eventRaiserMock = repository.StrictMock<SerializationEventRaiser>();
+      var eventRaiserMock = new Mock<SerializationEventRaiser> (MockBehavior.Strict);
+      eventRaiserMock.Setup (_ => _.InvokeAttributedMethod (instance, typeof (OnDeserializedAttribute), context)).Verifiable();
+      eventRaiserMock
+          .Protected()
+          .Setup<IReadOnlyCollection<MethodInfo>> ("FindDeserializationMethodsWithCache", typeof (ClassWithDeserializationEvents), typeof (OnDeserializedAttribute))
+          .Returns (new List<MethodInfo>());
 
-      eventRaiserMock.InvokeAttributedMethod (instance, typeof (OnDeserializedAttribute), context);
-      LastCall.CallOriginalMethod (OriginalCallOptions.CreateExpectation);
+      Assert.That (
+          () => PrivateInvoke.InvokeNonPublicMethod (eventRaiserMock.Object, "FindDeserializationMethodsWithCache", typeof (ClassWithDeserializationEvents), typeof (OnDeserializedAttribute)),
```

Comment:
why use a delegate here (=exception syntax)? 

make the invocation in a separate statement and assert the result. in general, I only do the inline call if it's short  enough to not threaten a linebreak.